src/backend/access/external/libchurl.c

```diff
@@ -312,16 +313,29 @@ CHURL_HANDLE churl_init(const char* url, CHURL_HEADERS headers)
 
 	/* needed to resolve localhost */
 	if (strstr(url, LocalhostIpV4) != NULL) {
-		struct curl_slist *resolve_hosts = NULL;
-		char *pxf_host_entry = (char *) palloc0(strlen(pxf_service_address) + strlen(LocalhostIpV4Entry) + 1);
-		strcat(pxf_host_entry, pxf_service_address);
-		strcat(pxf_host_entry, LocalhostIpV4Entry);
-		resolve_hosts = curl_slist_append(NULL, pxf_host_entry);
-		set_curl_option(context, CURLOPT_RESOLVE, resolve_hosts);
-		pfree(pxf_host_entry);
-	}
+		//get loopback interface ip address
+		char* loopback_addr = get_loopback_ip_addr();
+		elog(DEBUG1, "Loopback interface IP address: %s", loopback_addr);
+		//find host start position in url
+		char* start = strstr(url, LocalhostIpV4);
```

Comment:
Not double check but move the allocation of `start` into the above `if` statement so we don't need to call the function twice